File: Referate/algorithm.cpp
#include <iostream>
#include <vector>
using namespace std;

// Calcuelaza al k-lea numar fibbonacci (modulo 1000000)
int Fibb(int n)
{
    if (n <= 1)
        return n;
    return (Fibb(n - 1) + Fibb(n - 2)) % 1000000000;
}


// Numar de numere prime in intervalul 2-n
int CountPrimes(int n)
{
    vector <bool> ciur(n + 1);
    int raspuns = 0;
    for (int i = 2; i <= n; i++) {
        if (!ciur[i]) {
            for (int j = i; j <= n; j += i)
                ciur[j] = 1;
            raspuns++;
        }
    }
    return raspuns;
}

int main()
{

    cout << CountPrimes(20000000) << '\n';
    // cout << Fibb(40) << '\n';
    return 0;
}

File: Referate/algorithm.py
# Calcuelaza al k-lea numar fibbonacci (modulo 1000000)
def Fibb(n: int):
    if n <= 1:
        return n
    return (Fibb(n - 1) + Fibb(n - 2)) % 1000000000

# Numar de numere prime in intervalul 2-n
def CountPrimes(n: int):
    ciur = [False for i in range(n + 1)]
    raspuns = 0
    for i in range(2, n + 1):
        if not ciur[i]:
            for j in range(i, n + 1, i):
                ciur[j] = True
            raspuns += 1
    return raspuns

print(CountPrimes(30000000))
# print(Fibb(40))